Mage/src/main/java/mage/abilities/effects/common/combat/CantAttackYouUnlessPayLifeAllEffect.java

```diff
@@ -0,0 +1,74 @@
+package mage.abilities.effects.common.combat;
+
+import mage.abilities.Ability;
+import mage.abilities.costs.Cost;
+import mage.abilities.effects.PayCostToAttackBlockEffectImpl;
+import mage.constants.Duration;
+import mage.constants.Outcome;
+import mage.filter.common.FilterCreaturePermanent;
+import mage.game.Game;
+import mage.game.events.GameEvent;
+import mage.game.permanent.Permanent;
+
+/**
+ * @author PurpleCrowbar
+ */
+public class CantAttackYouUnlessPayLifeAllEffect extends PayCostToAttackBlockEffectImpl {
```

Comment:
Given that this class is identical co `CantAttackYouUnlessPayManaAllEffect`, I think it would make more sense to make `CantAttackYouUnlessPayManaAllEffect` more generic:
1. Change `CantAttackYouUnlessPayManaAllEffect` to `CantAttackYouUnlessPayAllEffect`. And change its `ManaCosts manaCosts` parameter change to `Cost cost`.
2. Get rid of `PayCostToAttackBlockEffectImpl(Duration duration, Outcome outcome, RestrictType restrictType, ManaCosts manaCosts)` and keep only the constructor that takes in `Cost cost`.
3. Inside the `Cost cost` constructor, do a `instanceof` check on the `cost` to see if to assign it to `this.cost` or `this.manaCosts`

Reply:
Done